### PR TITLE
RCA resource files

### PR DESCRIPTION
**Pull request by gvarao**

### CODE REVIEW

**Comment by gvarao** [comment]
jenkins please retry a build

**Comment by gvarao** [comment]
jenkins please retry a build

**Comment by gvarao** [comment]
jenkins please retry a build

**Comment by sharanyabhat** [comment]
jenkins please retry a build

**Comment by gvarao** [comment]
jenkins please retry a build

**Comment by gvarao** [comment]
jenkins please retry a build

**Review by gaddevijay — CHANGES_REQUESTED** [review]

File: juniper_official/Chassis/chassis.resource [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            field fan-name {
+                type string;
+                source {
+                    rule chassis.fan/check-fan-health {
+                        field-name name;
+                    }
+                }
+            }
+        }
+        resource chassis-temperature {
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Chassis/chassis.resource:15]: Chassis-temperature depended on fan failure hence not other way around. 
Agree with you and I have add depends-on-multiple-instances true;

File: juniper_official/Chassis/chassis.resource [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    locate-resource chassis/fan {
+                        label-as fan;
+                        where {
+                            matches-with "true" "true";
+                        }
+                    }
+                }
+            }
+        }
+        resource re-temperature {
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Chassis/chassis.resource:37]: Agree and modified to routing-engine

File: juniper_official/Interfaces/check-ifl-status.rule [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                type string;
+                description "Interfaces to be monitored";
+            }
+            trigger ifl-status {
+                synopsis "Sub-interface status check kpi";
+                description "Sets health based on ifl state";
+                frequency 1offset;
+                term ignore-ifl-admin-down {
+                    when {
+                        matches-with "$ifl-admin-status" DOWN {
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Interfaces/check-ifl-status.rule:51]: Initially i wanted to ignore admin down IFLs. But thought about and removed term now as it is not proper use case

File: juniper_official/Protocols/Isis/isis.resource [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                    }
+                }
+                source {
+                    rule protocol.isis/check-isis-statistics {
+                        field-name interface-name;
+                    }
+                }
+            }            
+            depends-on interfaces/sub-interface {
+                term ifl-dependency {
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Protocols/Isis/isis.resource:28]: We can form but it is not straight forward. I will try to add in next release

File: juniper_official/Protocols/Mpls/mpls.resource [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            field lsp-name {
+                type string;
+                source {
+                    rule protocol.mpls/check-lsp-state-oc {
+                        field-name lsp-name;
+                    }
+                }
+            }
+            depends-on protocols/rsvp {
+                term rsvp-neighbor-dependency {
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Protocols/Mpls/mpls.resource:23]: Yes we can have only one rsvp neighbor.

File: juniper_official/Protocols/Ospf/ospf.resource [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                        where {
+                            matches-with "$ospf-interface-split-1" "$ifl:interface-name";
+                            matches-with "$ospf-interface-split-2" "$ifl:sub-interface-index";
+                        }
+                    }
+                }
+                term ifl-remote-dependency {
+                	for-every-device {
+            			label-as remote-device;
+        			}
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Protocols/Ospf/ospf.resource:47]: Corrected

File: juniper_official/System/system.resource [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+                        field-name routing-engine;
+                    }
+                }
+                source {
+                    rule system.cpu.memory/check-system-cpu-memory-snmp {
+                        field-name description;
+                    }
+                }          
+            }
+            depends-on system/system-storage {
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/System/system.resource:49]: We have verified again with rama and he suggested.

File: juniper_official/Protocols/Ospf/ospf.resource [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        			    where {
+        			        matches-with "$local-routing-instance:router-id" "$remote-ospf:neighbor-id";
+        			        matches-with "$dr-address" "$remote-ospf:dr-address";
+        			    }
+        			}
+        			locate-resource remote-device:interface/interface {
+        			    label-as remote-if1;
+        			    with-capture-group ospf-remote-interface-split {
+                        	expression "(.*)\.(\d+)";
+                        	field-name "$remote-ospf:interface-name";
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Protocols/Ospf/ospf.resource:65]: Modified as we discussed

File: juniper_official/System/system.resource (hunk 7 above)
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/System/system.resource:49]: Modified as we discussed

File: juniper_official/System/system.resource [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+            }
+            field pid {
+                type string;
+                source {
+                    rule system.processes/check-process-memory{
+                        field-name pid;
+                    }
+                }
+            }
+            depends-on system/system-memory {
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/System/system.resource]: Modified as we discussed

File: juniper_official/Protocols/Ospf/ospf.resource [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+                source {
+                    rule protocol.ospf/check-ospf-neighbor-information {
+                        field-name interface-name;
+                    }
+                }
+            }
+            field neighbor-id {
+                type string;
+                source {
+                    rule protocol.ospf/check-ospf-neighbor-information {
```
[the remark this answers is not in the document]
Comment (gaddevijay) [reply juniper_official/Protocols/Ospf/ospf.resource:17]: Checked with Venkat. Seems like there is only one rule

File: juniper_official/Protocols/Ospf/ospf.resource (hunk 8 above)
[the remark this answers is not in the document]
Comment (gaddevijay) [reply juniper_official/Protocols/Ospf/ospf.resource:68]: This is not modified. Backend support is required

**Review by gaddevijay — APPROVED** [review]

**Pull request merged**